snapcraft/projects.py

```diff
@@ -442,6 +452,14 @@ def _validate_plugs(cls, plugs):
                 if isinstance(plug, list):
                     raise ValueError(f"Plug '{plug_name}' cannot be a list.")
 
+                if isinstance(plug, dict) and plug.get("default-provider"):
```

Comment:
`plugs` can be a dict of `ContentPlug` which was already validated by pydantic, so checking it again here is unnecessary?

Reply:
I see it can be Any `plugs: Optional[Dict[str, Union[ContentPlug, Any]]]`. Not sure what it could be.